pulsar-common/src/main/java/org/apache/pulsar/common/topics/TopicList.java

```diff
@@ -47,13 +47,16 @@ public static List<String> filterTopics(List<String> original, String regex) {
     }
     public static List<String> filterTopics(List<String> original, Pattern topicsPattern) {
 
-        final Pattern shortenedTopicsPattern = topicsPattern.toString().contains(SCHEME_SEPARATOR)
-                ? Pattern.compile(SCHEME_SEPARATOR_PATTERN.split(topicsPattern.toString())[1]) : topicsPattern;
+        final Pattern shortenedTopicsPattern = Pattern.compile(removeTopicDomainSchema(topicsPattern.toString()));
```

Comment:
It's interesting. @poorbarcode Do you know why we need to remove the topic domain scheme?

Reply:
Because we want to support that whether use set Topic domain schema or not 😂 , so just compare after removing the Topic domain schema.